### ISSUE TITLE
suddenly need to manually set context

### ISSUE THREAD

**Issue by marafa-sugarcrm** [kontena/lens#7826]
**Describe the bug**
was not able to connect to any cluster until i went to my cli and set a context using `kubectx $cluster_name`

**To Reproduce**

0. start lens
1. Go to clusters page
2. Click on any cluster
4. See error `Failed to start connection: ValidationError: "current-context" is not allowed to be empty`

**Expected behavior**

**Screenshots**
![Screenshot 2023-06-01 at 10 35 11 AM](https://github.com/lensapp/lens/assets/69474017/325a58e8-7a72-40d5-8106-611ccc4e25da)

**Environment (please complete the following information):**
- Lens Version: OpenLens: 6.5.2+286
- OS: [e.g. OSX] 
- Installation method (e.g. snap or AppImage in Linux): brew

**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:
```
Your logs go here...
```

**Kubeconfig:**
Quite often the problems are caused by malformed kubeconfig which the application tries to load. Please share your kubeconfig, remember to remove any secret and sensitive information.
```
your kubeconfig here
```

**Additional context**
after setting context, connecting to the cluster in openlens, unsetting context, i am still able to browse the cluster

**Comment by iamhritikkumar** [comment]
After upgrading Lens to  2023.5.310801-latest version, now facing the same issue.

**Comment by ps68060** [comment]
I had the same issue on Win10 after updating (01-Jun-2023) so I rolled back to the last download I had.

**Comment by rhalat** [comment]
I had a similar issue with latest version, but despite setting the context using `kubectx`, I continued to receive the following error message:

Failed to start connection: ValidationError: "contexts[4].context.cluster" is not allowed to be empty. "contexts[4].context.user" is not allowed to be empty. "contexts[5].context.cluster" is not allowed to be empty. "contexts[5].context.user" is not allowed to be empty

so I also rolled back to the last download I had, very old one, but it's working.

**Comment by iamhritikkumar** [comment]
@rhalat I have deleted the older version lens from my system if it's possible can you share the link from where I can download the older version lens dmg

**Comment by rhalat** [comment]
Yesterday, the older version I had was working, but today I'm getting the error message "This version of Lens Desktop is No Longer Maintained! Please Upgrade" and there's nothing I can do about it. So I guess it wouldn't help you much either.

**Comment by marafa-sugarcrm** [comment]
you can find all the previous releases here: https://github.com/MuhammedKalkan/OpenLens/releases (its merely a build repo)

**Comment by kapelan** [comment]
Any idea what is a cause? In my case switching between contexts does not help at all.

**Comment by lokeshm89** [comment]
I had the same issue, I deleted the empty configuration created by docker desktop from the config file and everything started working fine.

**Comment by lok1989** [comment]
> Any idea what is a cause? In my case switching between contexts does not help at all.

@[lokeshm89](https://github.com/lokeshm89)
lokeshm89 commented [now](https://github.com/lensapp/lens/issues/7826#issuecomment-1588103825)
I had the same issue, I deleted the empty configuration created by docker desktop from the config file and everything started working fine.

**Comment by cbown75** [comment]
I had an empty context in my config.  Once I removed that the errors when away.  Thanks for the pointer @lokeshm89

**Comment by marafa-sugarcrm** [comment]
i wish!
this didnt work for me. i recreated my config file and i still see this issue

fyi its `current-context: ""` that kills lens

**Comment by aignatovich** [comment]
After Lens decided to update itself similar issue was encountered.

Lens compained about `contexts` section of `kubeconfig`, particularly about a context having an empty user, like below:

```
- context:
    cluster: <cluster name>
    user: "" <-- Here is the issue
```

Removing stale context entry without a username solved the issue for me.

**Comment by marafa-sugarcrm** [comment]
the latest update did NOT fix this issue

NOTE: there seems to be 2 related issues here. 

1.  where there is an empty context `"contexts[5].context.cluster" is not allowed to be empty.` - this is due to a corrupt local .kube/config file and easily fixed 
2. where there is an empty `current-context` - this is due to lens - this github issue was opened for this

**Comment by brunolimadevops** [comment]
I had the same issue after update my lens about 7 days ago.
In my case, the URL of default cluster was `""`. I have changed change this line from `server: " "` to `server: literaly_some_word` and worked fine for me.

**Comment by marafa-sugarcrm** [comment]
@brunolima1989 thanks for the hint

i changed my ~/.kube/config/ from `current-context: ""` to `current-context: " "` and it worked.

i am guessing the k8s library used by lens was "updated" and broke things

**Comment by brunolimadevops** [comment]
Yes, it's possible. In k9s and aptakube (other 2 managers that I like a lot) the problem does not happen.

**Comment by belissner** [comment]
> @brunolima1989 thanks for the hint
> 
> i changed my ~/.kube/config/ from `current-context: ""` to `current-context: " "` and it worked.
> 
> i am guessing the k8s library used by lens was "updated" and broke things

This was a workaround for me as well. Or just delete the line: `current-context: ""`

**Comment by aomegax** [comment]
> > @brunolima1989 thanks for the hint
> > i changed my ~/.kube/config/ from `current-context: ""` to `current-context: " "` and it worked.
> > i am guessing the k8s library used by lens was "updated" and broke things
> 
> This was a workaround for me as well. Or just delete the line: `current-context: ""`

this is not a workaround for me 🤒

**Comment by tarsisazevedo** [comment]
I had some misconfigured contexts with `user` field empty. Just removed them from `~/.kube/config` and it worked.

**Comment by davebullock-ujet** [comment]
this drove me a bit crazy, would be good to just have Lens ignore config that isn't properly formatted or complete.

**Comment by LuWang1983** [comment]
To address the error `Failed to start connection: ValidationError: "contexts[4].context.cluster" is not allowed to be empty.`
Open `~/.kube/config`,  you may see something like this in your config file (see screenshot), delete line 7-9 or any field that's empty and you are not using and save. It worked for me afterwards. You don't have to restart Lens. 
![Screen Shot 2023-07-29 at 7 23 05 PM](https://github.com/lensapp/lens/assets/63315158/486567a2-b30d-48ce-a629-0140dac0c7af)

**Comment by iamhritikkumar** [comment]
In this latest version `Lens: 2023.8.31525-latest` they have fixed this issue

**Comment by abhishek6s** [comment]
for `Failed to start connection: ValidationError: "current-context" is not allowed to be empty`

1. go to $HOME/.kube/config
2. vi config
3. search for `current-context`
4. you will find this - >  current-context: ""
5. change it to any context value:  current-context: "context1" (you can find contexts in the same file)
6. Done

**Comment by wasuaje** [comment]
If you find a block like this: remove it
```
- context:
      cluster: ""
      user: ""
    name: default
```

**Comment by ro0NL** [comment]
same :roll_eyes: 

> you will find this - > current-context: ""
> change it to any context value: current-context: "context1" (you can find contexts in the same file)
> Done

yes, but who changes the current-context value constantly?! it's annoying.
